File: server/job/src/main/java/io/iamcyw/ams/job/strategy/persistence/StrategyPushMetaPO.java
package io.iamcyw.ams.job.strategy.persistence;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.*;

@Entity(name = "ALARM_STRATEGY_PUSH_META")
public class StrategyPushMetaPO extends PanacheEntity {

    @Column(name = "STRATEGY_PUSH_META_KEY")
    public String metaKey;

    @Column(name = "STRATEGY_PUSH_META_VALUE")
    public String metaValue;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "PUSH_ID")
    public StrategyPushPO push;

}

File: domain/src/jvmMain/kotlin/io/iamcyw/ams/domain/job/strategy/StrategyPushMetaPO.kt
package io.iamcyw.ams.domain.job.strategy

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity(name = "ALARM_STRATEGY_PUSH_META")
actual class StrategyPushMetaPO : PanacheEntityBase {

    companion object : PanacheCompanion<StrategyPushMetaPO>

    actual var metaKey: String = ""

    actual var metaValue: String = ""

    @get:ManyToOne(fetch = FetchType.LAZY)
    @get:JoinColumn(name = "PUSH_ID")
    actual var push: StrategyPushPO? = null

}